### PR TITLE
Begin implementing cooking proficiencies

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
Content "Adding cooking proficiencies to recipes, part 1"

#### Purpose of change
#49221 added cooking proficiencies for things that are currently representable with proficiencies, but they do nothing as yet

#### Describe the solution
This adds proficiencies to (at the time of writing) about half of the recipes in recipes-food.json, a disgustingly huge file that should be broken up into multiple files, but I am too busy for that.

the food_prep and knife_skills proficiencies may need some review here. Both add a very small time benefit when you learn them, so I have put them in recipes where I think any amount of organization and skill would help. In general, I assume that if a knife is involved, being good with it will help, and if more than one ingredient is involved, being able to sort stuff out and organize your recipe quickly will help. This is aided by the fact that most truly easy recipes are very quick, so in general food_prep is only going to offer a little boost, so if I accidentally put it on something that's a bit too simple, it won't have a big impact on gameplay.

#### Describe alternatives you've considered
The proficiency structure and choices could be different.
There are more proficiencies I'd like to have here, but for now with the boolean fail state I don't want them. Someday when we can determine if you have made yucky food, we'll need a proficiency series based on your ability to properly season complex stuff, but these proficiencies are just for "did you mess up or not".

#### Testing
wip

### CODE REVIEW

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -416,6 +443,7 @@
     "autolearn": true,
     "qualities": [ { "id": "CUT", "level": 1 }, { "id": "COOK", "level": 2 } ],
```
Comment (Maleclypse) [inline data/json/recipes/recipe_food.json:444]:
```suggestion
    "qualities": [ { "id": "COOK", "level": 2 } ],
```
Cooking oil from seeds doesn't use knives.

File: data/json/recipes/recipe_food.json [hunk 2]
```diff
@@ -416,6 +443,7 @@
...
     "tools": [ [ [ "surface_heat", 10, "LIST" ] ] ],
+    "proficiencies": [ { "proficiency": "prof_food_prep" }, { "proficiency": "prof_knife_skills" } ],
```
Comment (Maleclypse) [inline data/json/recipes/recipe_food.json:446]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_food_prep" } ],
```
Cooking oil

File: data/json/recipes/recipe_food.json [hunk 3]
```diff
@@ -1055,6 +1127,7 @@
     "autolearn": true,
     "qualities": [ { "id": "COOK", "level": 2 } ],
```
Comment (Maleclypse) [inline data/json/recipes/recipe_food.json:1128]:
```suggestion
    "qualities": [ { "id": "CUT", "level": 1 }, { "id": "COOK", "level": 2 } ],
```
Dogfood would need cut ingredients.

File: data/json/recipes/recipe_food.json [hunk 4]
```diff
@@ -1055,6 +1127,7 @@
...
     "tools": [ [ [ "surface_heat", 6, "LIST" ] ] ],
+    "proficiencies": [ { "proficiency": "prof_food_prep" } ],
```
Comment (Maleclypse) [inline data/json/recipes/recipe_food.json:1130]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_food_prep" }, { "proficiency": "prof_knife_skills" } ],
```
Dogfood

File: data/json/recipes/recipe_food.json [hunk 5]
```diff
@@ -1073,6 +1146,7 @@
     "autolearn": true,
     "qualities": [ { "id": "CUT", "level": 1 }, { "id": "COOK", "level": 2 } ],
     "tools": [ [ [ "surface_heat", 6, "LIST" ] ] ],
+    "proficiencies": [ { "proficiency": "prof_food_prep" } ],
```
Comment (Maleclypse) [inline data/json/recipes/recipe_food.json:1149]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_food_prep" }, { "proficiency": "prof_knife_skills" } ],
```
Catfood knife skills

**Review by Maleclypse — COMMENTED** [review]
I made it through catfood line 1151.   I'll try and come back to this when my will to live is stronger.

**Comment by I-am-Erk** [comment]
Feel like coming through and finishing that review? I'd like to merge this and move on with my life.

**Comment by Maleclypse** [comment]
> Feel like coming through and finishing that review? I'd like to merge this and move on with my life.

Oh wow. Sorry. Yeah I’ll do that.

**Review by Maleclypse — APPROVED** [review]
The rest of it looks good.    Thanks!

**Pull request merged**